referrer-policy/generic/referrer-policy-test-case.js

```diff
@@ -0,0 +1,144 @@
+var ReferrerPolicyTestCase = function(scenario, testDescription) {
+  // Check if scenario is valid.
+  test(function() {
+    // We extend the exsiting test_expansion_schema not to kill performance by
+    // copying.
+    var expectedFields = SPEC_JSON["test_expansion_schema"];
+    expectedFields["referrer_policy"] = SPEC_JSON["referrer_policy_schema"];
+
+    for (var field in expectedFields) {
+      assert_own_property(scenario, field,
+                          "The scenario contains field " + field)
+      assert_in_array(scenario[field], expectedFields[field],
+                      "Scenario's " + field + " is one of: " +
+                      expectedFields[field].join(", ")) + "."
+    }
+
+    // Check if the protocol is matched.
+    assert_equals(scenario["source_protocol"] + ":", location.protocol,
+                  "Protocol of the test page should match the scenario.")
+
+  }, "[ReferrerPolicyTestCase] The test scenario is valid.");
+
+  var subresourceInvoker = {
+    "a-tag": queryLink,
+    "area-tag": queryAreaLink,
```

Comment:
This actually lists all supported sub-resources and their handlers, however some of them get suppressed by the ```spec.src.json```. I would keep them here for now to still have support for extending when needed, but I might refactor this a bit.

With the spec JSON we can reduce the number of different sub-resources we test for in particular circumstances. See more on the ```excluded_tests``` section.

I am also concerned with the performance of the test suite as it currently runs for ~ 4 minutes on my machine. There will definitely be tests which we should omit.